src/Messages/AbstractMessage.php

```diff
@@ -125,4 +126,17 @@ public function setMessageId(string $messageId)
     {
         $this->messageId = $messageId;
     }
+
+    /**
+     * Return a Status response based on the information of the notification.
+     *
+     * @return \EC\Poetry\Messages\Responses\Status
+     */
+    public function generateResponse()
```

Comment:
We should have such a method not on the generic abstract message object but on the notification handler class, since in this way we kind of link any message to the status response message, which is really something we don't want to do it :D